pkg/test/integration/common/helpers/handling_files.go

```diff
@@ -36,61 +37,53 @@ func ParseK8sYaml(filepath string) ([]runtime.Object, []*schema.GroupVersionKind
 	acceptedK8sTypes := regexp.MustCompile(`(Role|ClusterRole|RoleBinding|ClusterRoleBinding|ServiceAccount|CustomResourceDefinition|kind: Deployment)`)
 	acceptedMCMTypes := regexp.MustCompile(`(MachineClass|Machine)`)
 	fileAsString := string(fileR[:])
-	sepYamlfiles := strings.Split(fileAsString, "---\n")
-	retObj := make([]runtime.Object, 0, len(sepYamlfiles))
-	retKind := make([]*schema.GroupVersionKind, 0, len(sepYamlfiles))
+	retObj := make([]runtime.Object, 0, 1)
+	retKind := make([]*schema.GroupVersionKind, 0, 1)
 	var retErr error
 	crdRegexp, _ := regexp.Compile("CustomResourceDefinition")
-	for _, f := range sepYamlfiles {
-		if f == "\n" || f == "" {
-			// ignore empty cases
-			continue
+	if fileAsString == "\n" || fileAsString == "" {
```

Comment:
we should take note to add a unit-test to `ParseK8sYaml` in the future . This is a big function and it is difficult to judge changes.

Reply:
Ideally we should break it down to `ParseK8sCRD`, `ParseMachineResources` and `ParseOtherK8sResources` or something like this.